### ISSUE TITLE
Clear Private Data on exit

### ISSUE THREAD

**Issue by davidcollini** [brave/brave-ios#5036]
## Problem Description 
My cache, browsing history, recent searches, and downloaded files all stay in the browser unless I manually clear them
## Feature Overview 
There should be a toggle for clear private data that says manual, on exit, and after a certain period of time

## Design 
![B0598D43-10AD-4D0C-917E-388BF1F48BC0](https://user-images.githubusercontent.com/89056889/156051571-c85e5ad7-1ec1-4fad-bc2c-1ef6f79a1a18.jpeg)

## Implementation Details 
You’d be able to click on this option and go into a sub-menu and choose manual, on exit, or after a certain time

## User Experience 

1. Go to settings
2. Brave shields and Privacy
3. Click on button and pick manual, on exit, or after a certain time

### Additional information 
N/A

**Comment by iccub** [comment]
Hi, we have private browsing mode which does just this, it clears all data when you exit the private mode or kill the app.
Any reasons you do not use it to achieve this effect?

**Comment by davidcollini** [comment]
I don’t use private browsing only because I don’t want to clear my shields or cookies

**Comment by davidcollini** [comment]
If there were an exceptions list for clearing data per site, like the DuckDuckGo app has, I would use private browsing only mode

**Comment by iccub** [comment]
I see, thanks.
We considered adding something like clear data on close, or maybe a quick clear-all button similar to DDG, i'll bring up this ticket in our triage meeting

Exclude list from data-clear sounds like a smart feature too

**Comment by davidcollini** [comment]
This feature could also eliminate the need for Private Browsing only mode to be an option, since all it’s feature will exist in Clear Private Data

**Comment by iccub** [comment]
We will probably move away from this but one cool feature about our private browsing is that the browsing data is never persisted in our database, we use a special in-memory only database, stored in RAM.

In general we've heard more about people wanting to persist their data in private mode because this is what all other major browsers do.

So in the future it will most likely be private mode with persisted data with an option to clear it on close,
and for regular mode a delete-all button with quick access,
an exclude list would be nice for both of those

**Comment by davidcollini** [comment]
And for when picking exceptions, there should be a list that shows the sites you have data on, with the sites having the most data at the top

**Comment by RHWHT** [comment]
> We will probably move away from this but one cool feature about our private browsing is that the browsing data is never persisted in our database, we use a special in-memory only database, stored in RAM.
> 
> In general we've heard more about people wanting to persist their data in private mode because this is what all other major browsers do.
> 
> So in the future it will most likely be private mode with persisted data with an option to clear it on close, and for regular mode a delete-all button with quick access, an exclude list would be nice for both of those

Doesn't persisted data, not stored in RAM, defeat the purpose of private browsing?

**Comment by iccub** [comment]
@RHWHT partially true, but for some reason all other browsers in the market persist private data and then clear it if needed.
Keeping things in RAM has its own set of difficulties

Good on-device file encryption solves privacy concerns for this one

**Comment by RHWHT** [comment]
Thanks for the explanation. When other browsers started using this method on iOS, I assumed "device encryption makes it okay" was the reasoning but wasn't sure.

**Comment by MahdiNazemi** [comment]
> Hi, we have private browsing mode which does just this, it clears all data when you exit the private mode or kill the app. Any reasons you do not use it to achieve this effect?

I may be wrong, but I think using the private browsing mode forces you to use the Dark/Night mode, which may not be what the user wants.

**Comment by davidcollini** [comment]
> > Hi, we have private browsing mode which does just this, it clears all data when you exit the private mode or kill the app. Any reasons you do not use it to achieve this effect?
> 
> 
> 
> I may be wrong, but I think using the private browsing mode forces you to use the Dark/Night mode, which may not be what the user wants.

Nah, while I don't like private browsing changing everything to purple, I basically want a more expanded version of #893
